### ISSUE TITLE
deploy process should validate yaml before starting

### ISSUE THREAD

**Issue by rtprio** [uselagoon/lagoon#257]

**Comment by dasrecht** [comment]
I had the issue with the tls-acme and insecure for the routes. somehow my yaml linter is happy both ways. lagoon on the other hand is not.

```
Subject: [PATCH] do you even yml?

---
 .lagoon.yml | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/.lagoon.yml b/.lagoon.yml
index c5664d8..de8602b 100644
--- a/.lagoon.yml
+++ b/.lagoon.yml
@@ -21,5 +21,5 @@ environments:
     routes:
       - nginx:
         - "staging.freshjobs.ch":
-          tls-acme: 'true'
-          insecure: Redirect
\ No newline at end of file
+            tls-acme: 'true'
+            insecure: Redirect
\ No newline at end of file
```

**Comment by erikrs** [comment]
suggestion: validate and correct line endings. (CRLF to LF)

based on following experience:
Yesterday had some long troubleshooting session regarding .lagoon.env.$branch environment variables. Long story short: it seemed that my .lagoon.env.dev file (local env. is on Windows) had CRLF line endings.
- This resulted that each environment variable ended with char 13. 
- This resulted it weird situation, eg. the node.js redis library tests a port variable, which was set via an environment variable which was set via .lagoon.env. file (via 50-dotenv.sh). 
whether is is numerical via regex (ie. /^\d+$/.test(). It failed because latest character (13) wasn't.

Bottom line: suggestion for change request: is there a way to protect developers (in mixed environments) from making this mistake. Eg. pro actively do a CRLF to LF conversion of .lagoon.env... files ?

**Comment by rtprio** [comment]
Adding it here, so I remember: either truncate generated project at 63 characters or reject the build when it becomes that long.

**Comment by rtprio** [comment]
this is an old library, but should partly assist.; http://rx.codesimply.com/index.html

**Comment by seanhamlin** [comment]
Another reason why this would be useful

```
+++ cat .lagoon.yml
+++ shyaml get-value environments.production.cronjobs.2.service
Error: invalid path 'environments.production.cronjobs.2.service', missing key 'service' in struct.
```

The build managed to progress 58 seconds, and 1500 lines behind it was struck down with this. It would be great to do this up front.

**Comment by twardnw** [comment]
Since we already have python in the build images, https://github.com/23andMe/Yamale looks like it would be a good solution

**Comment by smlx** [comment]
yup for the record yamale is also the tool used by helm for yaml linting/validation

**Comment by tobybellwood** [comment]
The lagoon linter and build deploy tools now require YAML to be valid (and slightly correct) before deployment. It doesn't do a full schema check, but that's a separate issue.
